pkg/component/extensions/operatingsystemconfig/init/templates/scripts/init.tpl.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "> Prepare temporary directory for image pull and mount"
+tmp_dir="$(mktemp -d)"
+trap 'ctr images unmount "$tmp_dir" && rm -rf "$tmp_dir"' EXIT
+
+echo "> Pull gardener-node-agent image and mount it to the temporary directory"
+ctr images pull  "{{ .image }}"
+ctr images mount "{{ .image }}" "$tmp_dir"
+
+echo "> Copy gardener-node-agent binary to host ({{ .binaryDirectory }}) and make it executable"
+cp -f "$tmp_dir/gardener-node-agent" "{{ .binaryDirectory }}"
+chmod +x "{{ .binaryDirectory }}/gardener-node-agent"
+
+echo "> Bootstrap gardener-node-agent"
+"{{ .binaryDirectory }}/gardener-node-agent" bootstrap --config={{ .configFile }}
```

Comment:
Do we want to use `exec` here?

Reply:
Why?

Reply:
As this call does not need to return to the shell script because it is the last line, we can reuse the same process. Hence, we save a small amount of (kernel) resources and have a slightly increased startup speed (as no new process needs to be created).
Then again, this is only a very small optimisation that may not be relevant looking at the bigger picture.
I will leave it up to you to decide whether to implement it or not.